switch.py

```diff
@@ -30,34 +35,56 @@
     """Set up the Sonoff LAN Mode Switch platform."""
     host = config.get(CONF_HOST)
     name = config.get(CONF_NAME)
+    icon = config.get(CONF_ICON)
+    device_id = config.get(CONF_DEVICE_ID)
+    api_key = config.get(CONF_API_KEY)
 
-    async_add_entities([HassSonoffSwitch(hass, host, name)], True)
+    # todo: remove once we have packaged correctly
+    # Add path so we can load dependant component (pysonofflan) from custom_components directory
+    # this support side by side execution 
+    import sys
+    path = hass.config.path('custom_components/sonoff_lan_mode_r3')
+    if path not in sys.path:
+        sys.path.insert(0, path)
 
+    async_add_entities([HassSonoffSwitchR3(hass, host, name, icon, device_id, api_key)], True)
 
-class HassSonoffSwitch(SwitchDevice):
-    """Home Assistant representation of a Sonoff LAN Mode device."""
 
-    def __init__(self, hass, host, name):
-        from pysonofflan import SonoffSwitch
+class HassSonoffSwitchR3(SwitchDevice):
+    """Home Assistant representation of a Sonoff LAN Mode device."""
 
-        _LOGGER.setLevel(logging.DEBUG)
+    def __init__(self, hass, host, name, icon, device_id, api_key):
+            
+        from pysonofflan3 import SonoffSwitch
```

Comment:
Ahhh, I see - let's pick this up in the `pysonofflan` PR thread, as this doesn't feel ideal to me

Reply:
I've made some improvements, but please feel free to refine

Reply:
Agreed, happy to discuss.